### PR TITLE
Add Documentation for AgentChat 

### PR DESCRIPTION
**Pull request by victordibia**
## Why are these changes needed?

Add documentation and examples for AgentChat. 

- [x] Add landing page introducing AgentChat and the concepts of `Agents` and `Teams`  
- [x] Provide a migration guide in the Quick Start section, with an example of how to migrate the basic stock price plot example from autogen `0.2x` to AgentChat `0.4x`.
- [x] Improve current documentation on `Tool Use`, showing example on using `LangChain Tools` 
- [x] Add a set of  3 concrete examples implementing workflows with AgentChat 
   - [x] Travel Planning 
   - [x] Company Research and Report Generation (using web search and custom tools)  
   - [x] Academic Literature Review with Arxiv (using web search, and arxiv tools)

## Related issue number

#3633 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: python/packages/autogen-core/docs/src/agentchat-user-guide/index.md [hunk 1]
```diff
@@ -2,20 +2,43 @@
...
-      User Guide for AutoGen AgentChat, a framework for building multi-agent applications with AI agents.
+      User Guide for AgentChat, a high-level api for AutoGen
 ---
 
 # AgentChat
 
-```{warning}
-🚧 Under construction 🚧
-```
+AgentChat is a high-level framework for building multi-agent applications with AI agents, utilizing the underlying `autogen-core` package. For beginner users, AgentChat is the recommended starting point. For advanced users, `autogen-core` provides more flexibility and control over the underlying components.
```
Comment (gagb) [inline python/packages/autogen-core/docs/src/agentchat-user-guide/index.md:10]:
```suggestion
AgentChat is a high-level package for building multi-agent applications built on top of the `autogen-core` package. For beginner users, AgentChat is the recommended starting point. For advanced users, `autogen-core` provides more flexibility and control over the underlying components.
```

File: python/packages/autogen-core/docs/src/agentchat-user-guide/index.md [hunk 2]
```diff
@@ -2,20 +2,43 @@
...
+
+AgentChat aims to provide intuitive defaults, such as Agents with preset behaviors and Teams with predefined communication protocols, to simplify building multi-agent applications.
```
Comment (gagb) [inline python/packages/autogen-core/docs/src/agentchat-user-guide/index.md:12]:
```suggestion
AgentChat aims to provide intuitive defaults, such as **Agents** with preset behaviors and **Teams** with predefined communication protocols, to simplify building multi-agent applications.
```

File: python/packages/autogen-core/docs/src/agentchat-user-guide/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+``````
+
+AgentChat in v0.4x provides similar abstractions to the default agents in v0.2x:
+
+- `CodingAssistantAgent` (v0.4x) ~ `AssistantAgent` (v0.2x)
+- `CodeExecutorAgent` (v0.4x) ~ `UserProxyAgent` with code execution (v0.2x)
+
+Key differences:
+
+1. In v0.4x, agent interactions are managed by `Teams` (e.g., `RoundRobinGroupChat`), replacing direct chat initiation.
```
Comment (gagb) [inline python/packages/autogen-core/docs/src/agentchat-user-guide/quickstart.md:82]: I find it confusing that we use the term Team in text but the code uses GroupChat. Is there a plan to extend the Team beyond group chats? if so this text should clarify that and instead say `Teams` (e.g., implementation of group chat-based teams such as ...)

Comment (victordibia) [reply python/packages/autogen-core/docs/src/agentchat-user-guide/quickstart.md:82]: Yeah, I had the same thought while writing this ... I think it just might be reasonable to rename it to Team. 

Some other related thoughts. 

- Single `AssistantAgent` that can take params that govern behaviours e.g., `description`, `system_message`, `code_executor`, `tools_schema` ...  
- Single `Team` or `AgentTeam` that takes params that govern behaviours e.g. `max_turns`, `speaker_selection` (round_robin, llm, graph etc).

Comment (gagb) [reply python/packages/autogen-core/docs/src/agentchat-user-guide/quickstart.md:82]: Are there any teams that aren't considered group chats?
If yes, then group chat is more specific and accurate.
If no, renaming to Teams is okay with me.

Comment (afourney) [reply python/packages/autogen-core/docs/src/agentchat-user-guide/quickstart.md:82]: I'd like to chat a bit more about how tools are incorporated. We should have a design session with the (human) team on this.

Comment (ekzhu) [reply python/packages/autogen-core/docs/src/agentchat-user-guide/quickstart.md:82]: The resolution on tools is in #3665

**Comment by victordibia** [comment]
Note: I have merged this into a [branch](https://github.com/microsoft/autogen/compare/improve_docs_vd) to improve docs overall.

**Comment by victordibia** [comment]
@ekzhu , I think this is ready for review

**Review by ekzhu — APPROVED** [review]

**Pull request merged**